### PR TITLE
feat(skills): add autoresearch skill for autonomous research orchestration

### PR DESCRIPTION
**Pull request by howardleegeek**
## Summary

Add `autoresearch` skill for autonomous, continuous research with a two-loop architecture.

## What's New

- **Inner Loop**: Rapid experiment iteration with clear measurable outcomes
- **Outer Loop**: Periodic synthesis, pattern discovery, and direction setting
- **Research Workspace Templates**: State tracking, findings synthesis, decision log
- **Example Project**: LoRA rank convergence study demonstrating the workflow
- **Configuration Options**: Loop intervals, auto-commit, workspace settings

## Use Cases

- Literature surveys with iterative hypothesis testing
- Benchmark optimization (e.g., "make val_loss go down")
- Mechanistic interpretability studies
- Hyperparameter discovery

## Files Added

```
skills/research/autoresearch/
├── SKILL.md                           # Main skill documentation
├── templates/
│   ├── research-state.yaml            # State tracking template
│   ├── findings.md                    # Synthesis template
│   └── research-log.md                # Decision log template
└── examples/
    ├── README.md                      # Example overview
    └── lora-rank-study.md             # Example research project
```

## Testing

- [x] Skill YAML frontmatter validated
- [x] Templates render correctly
- [x] Example follows autoresearch methodology
- [x] DESCRIPTION.md updated with skill overview

## Related Skills

Integrates with: `arxiv`, `research-paper-writing`, `web-search`, `notebooklm`

## Documentation

Full documentation in `SKILL.md` including:
- Two-loop architecture explanation
- Workspace structure
- Research discipline guidelines
- Configuration options
- Best practices

---

**Inspired by:** Andrej Karpathy's autoresearch methodology
**Compatible with:** agentskills.io open standard

### CODE REVIEW

File: skills/research/autoresearch/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
+---
+name: autoresearch
+description: Autonomous research orchestration for AI coding agents. Run continuous, self-directed research with a two-loop architecture — rapid inner-loop experiments and periodic outer-loop synthesis. Ideal for literature surveys, hypothesis testing, benchmark optimization, and iterative discovery. No human hand-holding required.
```
Comment (teknium1) [inline skills/research/autoresearch/SKILL.md:3]: This description exceeds the repository's hardline 60-character limit and is not a single sentence. Please reduce it to a concise, trigger-focused sentence ending in a period (`AGENTS.md:888-900`).

File: skills/research/autoresearch/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+version: 1.0.0
+author: Hermes Agent
+license: MIT
+metadata:
+  hermes:
+    tags: [Research, Autonomous, Experiments, ML, AI, Literature, Hypothesis, Benchmark, Optimization]
```
Comment (teknium1) [inline skills/research/autoresearch/SKILL.md:9]: `web-search` and `notebooklm` do not resolve to tracked in-repo or optional skills on current HEAD. Please retain only resolvable related skills or explicitly document an external prerequisite (`AGENTS.md:926-931`).

File: skills/research/autoresearch/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    related_skills: [arxiv, research-paper-writing, web-search, notebooklm]
+    config:
```
Comment (teknium1) [inline skills/research/autoresearch/SKILL.md:11]: This mapping is not the supported skill-config schema. `extract_skill_config_vars()` expects a list of objects containing `key` and `description` (`agent/skill_utils.py:622-678`), so this declaration is silently ignored.

File: skills/research/autoresearch/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+Ideal for literature surveys, hypothesis testing, benchmark optimization, mechanistic interpretability studies, and any research requiring iterative experimentation.
+
+## When to Use
+
+| Scenario | Use Autoresearch? |
+|----------|-------------------|
+| "I want to explore X and see what works" | ✅ Yes |
+| "Does technique Y improve metric Z?" | ✅ Yes |
+| "What's the state of the art for problem W?" | ✅ Yes (bootstrap + literature) |
+| "Train a model with specific hyperparameters" | ❌ Use domain skills directly |
```
Comment (teknium1) [inline skills/research/autoresearch/SKILL.md:40]: `research_init` has no tool or handler on current HEAD. A skill can be invoked through its own generated `/autoresearch` slug, but it cannot create this API; remove this call or implement a supported interface.

File: skills/research/autoresearch/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+Always commit your protocol to git BEFORE executing:
+
+```bash
+git add experiments/H001-protocol.md
+git commit -m "research(protocol): H001 — cosine warmup improves convergence"
+# THEN run the experiment
+```
+
+This creates temporal proof your plan existed before results.
```
Comment (teknium1) [inline skills/research/autoresearch/SKILL.md:119]: Only `/autoresearch` can be supplied dynamically as this skill's slug. Current HEAD has no handlers for the listed `/research-*` commands; please remove these claims or add the corresponding supported command implementation.

**Review by teknium1 — COMMENTED** [review]
Thanks for the self-contained research-workflow contribution. The templates are potentially reusable, but the skill currently advertises interfaces Hermes does not provide.

### Problems
- `skills/research/autoresearch/SKILL.md:11-20` uses a keyed `metadata.hermes.config` mapping. `agent/skill_utils.py:647-659` only accepts list items with `key` and `description`, so these settings are skipped; valid skill settings are stored beneath `skills.config.*` (`agent/skill_utils.py:720-754`).
- `SKILL.md:40` and `:119-124` promise `research_init` and `/research-*` commands absent from current HEAD. The existing dynamic mechanism supplies only the skill-slug command (`agent/skill_commands.py:1-4`, `cli.py:3555-3561`).
- The new skill also misses hardline repository standards in `AGENTS.md:888-950`: concise description, contributor-first author credit, resolvable related skills, and a `tests/skills/test_<skill>_skill.py` test.

### Suggested changes
- Use the supported config declaration schema and document `skills.config.autoresearch.*`.
- Remove unsupported commands/tool calls or implement them through existing surfaces.
- Modernize frontmatter, references, structure, and add the required skill test.

Automated hermes-sweeper review.
